include/s3select.h

```diff
@@ -1205,34 +1215,54 @@ void push_in_predicate::builder(s3select* self, const char* a, const char* b) co
   self->getAction()->inMainArg = 0;
 }
 
-void push_like_predicate::builder(s3select* self, const char* a, const char* b) const
+void push_like_predicate_no_escape::builder(s3select* self, const char* a, const char* b) const
 {
-  // expr like expr ; both should be string, the second expression could be compiled at the
-  // time AST is build, which will improve performance much.
 
   std::string token(a, b);
   std::string in_function("#like_predicate#");
 
   __function* func = S3SELECT_NEW(self, __function, in_function.c_str(), self->getS3F());
 
+  variable* v = nullptr;
```

Comment:
no need for that.

variable* v = S3SELECT_NEW(self, variable, s3select_reserved_word::reserve_word_en_t::S3S_NULL);